discord/bot.go

```diff
@@ -253,16 +268,19 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
 	}
 }
 
-func newGuild(s *discordgo.Session, m *discordgo.GuildCreate) {
-	log.Printf("Added to new Guild, id %s, name %s", m.Guild.ID, m.Guild.Name)
-	AllGuilds[m.ID] = &GuildState{
-		ID:                   m.ID,
-		delays:               GameDelays{},
-		VoiceStatusCache:     make(map[string]UserData),
-		voiceStatusCacheLock: sync.RWMutex{},
-		GameState:            game.GameState{Phase: game.UNINITIALIZED},
-		gameStateLock:        sync.RWMutex{},
-		Tracking:             make(map[string]Tracking),
-		TextChannelId:        "",
+func newGuild(moveDeadPlayers bool) func(s *discordgo.Session, m *discordgo.GuildCreate) {
```

Comment:
Ah clever, I like this format. For some reason always forget I can pass around functions, even as return types, for a function that also accepts a separate arg